spec/query_spec.rb

```diff
@@ -105,9 +105,14 @@ def test_method
   end
 
   describe '#ref' do
-    it 'creates a ref' do
-      ref = random_ref
-      expect(Fauna::Query.ref(ref)).to eq(Fauna::Ref.new(ref))
+    it 'returns a ref from a string' do
+      str = random_ref_string
+      expect(Fauna::Query.ref(str)).to eq(Fauna::Ref.new(str))
+    end
+
+    it 'constructs a ref' do
+      expect(client.query { ref(@test_class, '123') }).to eq(Fauna::Ref.new('classes/query_test/123'))
+      expect(client.query { ref(@test_class, next_id) }.value).to match(/^classes\/query_test\/\d+$/)
```

Comment:
I think doing that and picking exactly the syntax we want would be best. Just would need to spend time going through it's settings and locking in what we want it to do.